In [1]:
import numpy as np
import pandas as pd


In [2]:
user_data = pd.read_csv('ml-100k/u.data' , sep='\t', header=None , names=["uid", "movieid", "rating" ,"timestamp"])
user_data.describe()

uid        movieid         rating     timestamp
count  100000.00000  100000.000000  100000.000000  1.000000e+05
mean      462.48475     425.530130       3.529860  8.835289e+08
std       266.61442     330.798356       1.125674  5.343856e+06
min         1.00000       1.000000       1.000000  8.747247e+08
25%       254.00000     175.000000       3.000000  8.794487e+08
50%       447.00000     322.000000       4.000000  8.828269e+08
75%       682.00000     631.000000       4.000000  8.882600e+08
max       943.00000    1682.000000       5.000000  8.932866e+08

In [3]:
movies_data = pd.read_csv('ml-100k/u.item' , sep='|', header=None , names=["movieid", "title" ,"release_date", "video_realease_date", "imdb", "unknown", "action", "adventure", "animation", "children", "comedy", "crime", "documnetary", "drama", "fantasy", "noir", "horror", "musical", "mistery", "romance", "scifi", "thriller", "war", "western"])
movies_data.describe()

movieid  video_realease_date      unknown       action  \
count  1682.000000                  0.0  1682.000000  1682.000000   
mean    841.500000                  NaN     0.001189     0.149227   
std     485.695893                  NaN     0.034473     0.356418   
min       1.000000                  NaN     0.000000     0.000000   
25%     421.250000                  NaN     0.000000     0.000000   
50%     841.500000                  NaN     0.000000     0.000000   
75%    1261.750000                  NaN     0.000000     0.000000   
max    1682.000000                  NaN     1.000000     1.000000   

         adventure    animation     children       comedy        crime  \
count  1682.000000  1682.000000  1682.000000  1682.000000  1682.000000   
mean      0.080262     0.024970     0.072533     0.300238     0.064804   
std       0.271779     0.156081     0.259445     0.458498     0.246253   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.000000     0.000000     0.000000     0.000000     0.000000   
50%       0.000000     0.000000     0.000000     0.000000     0.000000   
75%       0.000000     0.000000     0.000000     1.000000     0.000000   
max       1.000000     1.000000     1.000000     1.000000     1.000000   

       documnetary  ...     fantasy         noir       horror      musical  \
count  1682.000000  ...  1682.00000  1682.000000  1682.000000  1682.000000   
mean      0.029727  ...     0.01308     0.014269     0.054697     0.033294   
std       0.169882  ...     0.11365     0.118632     0.227455     0.179456   
min       0.000000  ...     0.00000     0.000000     0.000000     0.000000   
25%       0.000000  ...     0.00000     0.000000     0.000000     0.000000   
50%       0.000000  ...     0.00000     0.000000     0.000000     0.000000   
75%       0.000000  ...     0.00000     0.000000     0.000000     0.000000   
max       1.000000  ...     1.00000     1.000000     1.000000     1.000000   

           mistery      romance        scifi     thriller          war  \
count  1682.000000  1682.000000  1682.000000  1682.000000  1682.000000   
mean      0.036266     0.146849     0.060048     0.149227     0.042212   
std       0.187008     0.354061     0.237646     0.356418     0.201131   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.000000     0.000000     0.000000     0.000000     0.000000   
50%       0.000000     0.000000     0.000000     0.000000     0.000000   
75%       0.000000     0.000000     0.000000     0.000000     0.000000   
max       1.000000     1.000000     1.000000     1.000000     1.000000   

           western  
count  1682.000000  
mean      0.016052  
std       0.125714  
min       0.000000  
25%       0.000000  
50%       0.000000  
75%       0.000000  
max       1.000000  

[8 rows x 21 columns]

# Data description

Total Movies: 1682
Total Users: 943
Total Ratings: 100 000

In [4]:
null_genre=movies_data.isnull().sum()
print(null_genre)

movieid                   0
title                     0
release_date              1
video_realease_date    1682
imdb                      3
unknown                   0
action                    0
adventure                 0
animation                 0
children                  0
comedy                    0
crime                     0
documnetary               0
drama                     0
fantasy                   0
noir                      0
horror                    0
musical                   0
mistery                   0
romance                   0
scifi                     0
thriller                  0
war                       0
western                   0
dtype: int64


In [5]:
null_users=user_data.isnull().sum()
print(null_users)

uid          0
movieid      0
rating       0
timestamp    0
dtype: int64


In [8]:
movies_data.drop('video_realease_date', axis=1, inplace=True)

In [9]:
movies_data.describe()

movieid      unknown       action    adventure    animation  \
count  1682.000000  1682.000000  1682.000000  1682.000000  1682.000000   
mean    841.500000     0.001189     0.149227     0.080262     0.024970   
std     485.695893     0.034473     0.356418     0.271779     0.156081   
min       1.000000     0.000000     0.000000     0.000000     0.000000   
25%     421.250000     0.000000     0.000000     0.000000     0.000000   
50%     841.500000     0.000000     0.000000     0.000000     0.000000   
75%    1261.750000     0.000000     0.000000     0.000000     0.000000   
max    1682.000000     1.000000     1.000000     1.000000     1.000000   

          children       comedy        crime  documnetary        drama  \
count  1682.000000  1682.000000  1682.000000  1682.000000  1682.000000   
mean      0.072533     0.300238     0.064804     0.029727     0.431034   
std       0.259445     0.458498     0.246253     0.169882     0.495368   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.000000     0.000000     0.000000     0.000000     0.000000   
50%       0.000000     0.000000     0.000000     0.000000     0.000000   
75%       0.000000     1.000000     0.000000     0.000000     1.000000   
max       1.000000     1.000000     1.000000     1.000000     1.000000   

          fantasy         noir       horror      musical      mistery  \
count  1682.00000  1682.000000  1682.000000  1682.000000  1682.000000   
mean      0.01308     0.014269     0.054697     0.033294     0.036266   
std       0.11365     0.118632     0.227455     0.179456     0.187008   
min       0.00000     0.000000     0.000000     0.000000     0.000000   
25%       0.00000     0.000000     0.000000     0.000000     0.000000   
50%       0.00000     0.000000     0.000000     0.000000     0.000000   
75%       0.00000     0.000000     0.000000     0.000000     0.000000   
max       1.00000     1.000000     1.000000     1.000000     1.000000   

           romance        scifi     thriller          war      western  
count  1682.000000  1682.000000  1682.000000  1682.000000  1682.000000  
mean      0.146849     0.060048     0.149227     0.042212     0.016052  
std       0.354061     0.237646     0.356418     0.201131     0.125714  
min       0.000000     0.000000     0.000000     0.000000     0.000000  
25%       0.000000     0.000000     0.000000     0.000000     0.000000  
50%       0.000000     0.000000     0.000000     0.000000     0.000000  
75%       0.000000     0.000000     0.000000     0.000000     0.000000  
max       1.000000     1.000000     1.000000     1.000000     1.000000

In [10]:
movies_data

movieid                                      title release_date  \
0           1                           Toy Story (1995)  01-Jan-1995   
1           2                           GoldenEye (1995)  01-Jan-1995   
2           3                          Four Rooms (1995)  01-Jan-1995   
3           4                          Get Shorty (1995)  01-Jan-1995   
4           5                             Copycat (1995)  01-Jan-1995   
...       ...                                        ...          ...   
1677     1678                          Mat' i syn (1997)  06-Feb-1998   
1678     1679                           B. Monkey (1998)  06-Feb-1998   
1679     1680                       Sliding Doors (1998)  01-Jan-1998   
1680     1681                        You So Crazy (1994)  01-Jan-1994   
1681     1682  Scream of Stone (Schrei aus Stein) (1991)  08-Mar-1996   

                                                   imdb  unknown  action  \
0     http://us.imdb.com/M/title-exact?Toy%20Story%2...        0       0   
1     http://us.imdb.com/M/title-exact?GoldenEye%20(...        0       1   
2     http://us.imdb.com/M/title-exact?Four%20Rooms%...        0       0   
3     http://us.imdb.com/M/title-exact?Get%20Shorty%...        0       1   
4     http://us.imdb.com/M/title-exact?Copycat%20(1995)        0       0   
...                                                 ...      ...     ...   
1677  http://us.imdb.com/M/title-exact?Mat%27+i+syn+...        0       0   
1678  http://us.imdb.com/M/title-exact?B%2E+Monkey+(...        0       0   
1679      http://us.imdb.com/Title?Sliding+Doors+(1998)        0       0   
1680  http://us.imdb.com/M/title-exact?You%20So%20Cr...        0       0   
1681  http://us.imdb.com/M/title-exact?Schrei%20aus%...        0       0   

      adventure  animation  children  comedy  ...  fantasy  noir  horror  \
0             0          1         1       1  ...        0     0       0   
1             1          0         0       0  ...        0     0       0   
2             0          0         0       0  ...        0     0       0   
3             0          0         0       1  ...        0     0       0   
4             0          0         0       0  ...        0     0       0   
...         ...        ...       ...     ...  ...      ...   ...     ...   
1677          0          0         0       0  ...        0     0       0   
1678          0          0         0       0  ...        0     0       0   
1679          0          0         0       0  ...        0     0       0   
1680          0          0         0       1  ...        0     0       0   
1681          0          0         0       0  ...        0     0       0   

      musical  mistery  romance  scifi  thriller  war  western  
0           0        0        0      0         0    0        0  
1           0        0        0      0         1    0        0  
2           0        0        0      0         1    0        0  
3           0        0        0      0         0    0        0  
4           0        0        0      0         1    0        0  
...       ...      ...      ...    ...       ...  ...      ...  
1677        0        0        0      0         0    0        0  
1678        0        0        1      0         1    0        0  
1679        0        0        1      0         0    0        0  
1680        0        0        0      0         0    0        0  
1681        0        0        0      0         0    0        0  

[1682 rows x 23 columns]